src/sqlancer/ProviderAdapter.java

```diff
@@ -52,7 +52,7 @@ public Reproducer<G> generateAndTestDatabase(G globalState) throws Exception {
 
                     } catch (AssertionError e) {
                         Reproducer<G> reproducer = oracle.getLastReproducer();
-                        if (reproducer != null) {
+                        if (reproducer.getReproducerState()) {
```

Comment:
Is there any reason why need this method? I think it would be better to check for `null`: either we have the reproducer, or it is `null`. Having a reproducer object that indicates that we don't have the actual state is a bit confusing, I think.

Reply:
`reproducer` is returned by each call of `check`, instead of bug only, so we need to distinguish the returned `reproduer` denotes a bug or not.